### PR TITLE
[IO][canutils]: add direction support

### PR DESCRIPTION
**Pull request by jazi007**
Closes https://github.com/hardbyte/python-can/issues/1243

### CODE REVIEW

File: can/io/canutils.py [hunk 1]
```diff
@@ -164,8 +170,16 @@ def on_message_received(self, msg):
         else:
             framestr += " %03X#" % (msg.arbitration_id)
 
+        if msg.is_error_frame:
+            eol = "\n"
+        else:
+            if msg.is_rx:
+                eol = " R\n"
+            else:
+                eol = " T\n"
```
Comment (felixdivo) [inline can/io/canutils.py:179]: It's shorter and still well-readable:

```suggestion
            eol = " R\n" if msg.is_rx else " T\n"
```

File: can/io/canutils.py [hunk 2]
```diff
@@ -164,8 +170,16 @@ def on_message_received(self, msg):
...
+
         if msg.is_remote_frame:
-            framestr += "R\n"
+            framestr += "R%s" % (eol)
```
Comment (felixdivo) [inline can/io/canutils.py:182]:
```suggestion
            framestr += f"R{eol}"
```

File: can/io/canutils.py [hunk 3]
```diff
@@ -174,6 +188,6 @@ def on_message_received(self, msg):
                 if msg.error_state_indicator:
                     fd_flags |= CANFD_ESI
                 framestr += "#%X" % fd_flags
-            framestr += "%s\n" % (msg.data.hex().upper())
+            framestr += "%s%s" % (msg.data.hex().upper(), eol)
```
Comment (felixdivo) [inline can/io/canutils.py:191]:
```suggestion
            framestr += f"{msg.data.hex().upper()}{eol}"
```

**Review by felixdivo — CHANGES_REQUESTED** [review]
Apart from the above, this looks good.

For reference: https://github.com/linux-can/can-utils/blob/master/asc2log.c#L185

File: can/io/canutils.py [hunk 4]
```diff
@@ -51,7 +51,12 @@ def __iter__(self) -> Generator[Message, None, None]:
                 continue
 
             channel_string: str
-            timestamp_string, channel_string, frame = temp.split()
+            if temp[-2:].lower() in [" r", " t"]:
```
[the remark this answers is not in the document]
Comment (jazi007) [reply can/io/canutils.py]: Done

File: can/io/canutils.py (hunk 1 above)
Comment (jazi007) [reply can/io/canutils.py]: Done

File: can/io/canutils.py (hunk 2 above)
Comment (jazi007) [reply can/io/canutils.py]: Done

File: can/io/canutils.py (hunk 3 above)
Comment (jazi007) [reply can/io/canutils.py]: Done

**Review by felixdivo — APPROVED** [review]

**Comment by felixdivo** [comment]
As [mentioned in issue \#1217](https://github.com/hardbyte/python-can/pull/1217#issuecomment-1055706495): This can be merged now.

**Pull request merged**